packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -100,7 +100,7 @@ export class Cluster extends Resource implements ICluster {
       throw new Error("Can only add default namespace once.");
     }
 
-    const namespaceType = options.type === undefined || options.type === NamespaceType.PRIVATE_DNS
+    const namespaceType = options.type === undefined
```

Comment:
If options.type is PRIVATE_DNS, then you'll actually be setting the namespaceType to DNS_PUBLIC even though it should be private. You can either leave it as is or change it to something like:

const namespaceType = options.type !== undefined 
? options.type
: cloudmap.NamespaceType.DNS_PRIVATE

Reply:
Yes, I agree with you. Nice catch!